src/generic_ui/scripts/ui.ts

```diff
@@ -190,11 +191,27 @@ module UI {
      * Upon construction, the UI installs update handlers on core.
      */
     constructor(
-        public core   :uproxy_core_api.CoreApi,
+        public core   :CoreConnector,
         public browserApi :BrowserAPI) {
       // TODO: Determine the best way to describe view transitions.
       this.view = ui_constants.View.SPLASH;  // Begin at the splash intro.
 
+      core.on('connect', () => {
```

Comment:
For those though, I'd imagine anything would be structured like `instance.on('connect', () => {})` or `network.on('connect', () => {})`.  The fact that we are looking at a `connect` event from the core connector seems clear enough about what has happened.